### PR TITLE
[rolling] Add master_board_sdk to distribution.yaml

### PR DESCRIPTION
**Pull request by olivier-stasse**
# Please Add This Package to be indexed in the rosdistro.

rolling

# The source is here:

https://github.com/open-dynamic-robot-initiative/master-board

We would like to start a release repository in ros2-gbp but for this we would like to know if the 
status of the repository is good enough, most notably the naming.

# Checks
 - [X] All packages have a declared license in the package.xml
 - [X] This repository has a LICENSE file
 - [X] This package is expected to build on the submitted rosdistro

### CODE REVIEW

File: rolling/distribution.yaml
Comment (tfoote) [inline rolling/distribution.yaml:2600]: This name is definitely too generic. If you read the name it would be expected that it would be a board for all use cases? I'd recommend a prefix `odri_` potentially or something related to the Solo Quadruped. 

Just reading the README opening "Hardware and Firmware of the Solo Quadruped Master Board."  It's optimal that the name capture that aspect. Maybe `solo_quad_master_board` or `odri_master_board` if it's intended to be more generic than the one robot.

File: rolling/distribution.yaml
Comment (tfoote) [inline rolling/distribution.yaml:2603]: All the urls should end in .git

File: rolling/distribution.yaml
Comment (tfoote) [inline rolling/distribution.yaml:2610]: This repo doesn't exist yet. Please remove the release stanza and ask for it to be created after this is reviewed and merged via: https://github.com/ros2-gbp/ros2-gbp-github-org/blob/latest/CONTRIBUTING.md

Then bloom can fill in the release stanza.

**Review by tfoote — CHANGES_REQUESTED** [review]

**Comment by sloretz** [comment]
Holding for an upcoming [Rolling sync](https://discourse.ros.org/t/preparing-for-rolling-sync-2023-09-27/33770)

**Comment by olivier-stasse** [comment]
Dear @tfoote thanks for the feedback.
This PR is now without the release part (if I understood correctly your point).
I did a release repository on another organisation
https://github.com/stack-of-tasks/odri_master_board_sdk_release
because my tentative to have a release repository on ros2-gbp and a release team is still being processed:
https://github.com/ros2-gbp/ros2-gbp-github-org/pull/337

**Comment by tfoote** [comment]
Thanks removing the release stanza is one of the things that I commented on. The comments on the url endings is still pending.  

You've renamed the repo entry here. But the package which you're trying to release does not yet appear to be similarly renamed

https://github.com/open-dynamic-robot-initiative/master-board/blob/master/sdk/master_board_sdk/package.xml#L3


We use this submission as an opportunity to review the names and licenses of the code. Once this is reviewed and merged, then you can request a release repository referencing this entry. And that will allow you to make formal releases.

**Comment by olivier-stasse** [comment]
> Thanks removing the release stanza is one of the things that I commented on. The comments on the url endings is still pending.
Yes sorry. This is fixed in the new version of the PR.

> You've renamed the repo entry here. But the package which you're trying to release does not yet appear to be similarly renamed
> 
> https://github.com/open-dynamic-robot-initiative/master-board/blob/master/sdk/master_board_sdk/package.xml#L3
Yes you are right. 

During private exchanges, the maintainer pointed out that they have a lot code depending on this current naming scheme. So the new PR point towards another repository with the naming scheme requested for ROS.
I hope that at some point this will be integrated directly in the upstream repository.
 
> We use this submission as an opportunity to review the names and licenses of the code. Once this is reviewed and merged, then you can request a release repository referencing this entry. And that will allow you to make formal releases.
Thanks for the clarification.

**Comment by olivier-stasse** [comment]
I would be interested by having a feedback on what is wrong with this PR, or the repository. I guess that the stale label comes partly due to ROSCon. @tfoote could you change it to persistent or just close it if you think we need to rework it/change the structure of the repository. Thanks for your time.

**Review by tfoote — APPROVED** [review]
Thanks for the renaming. I understand that the renaming is often hard if the project has already grown. 

Sorry, github flagged the PR as a new contributor so needed explicit approval to run. . And I think that this got dropped through the triage over ROSCon. 

The CI flagged that the stanza needs to be moved to keep alphabetical order with the rename. But after that this looks good to me.

**Comment by olivier-stasse** [comment]
Dear @tfoote thanks for the feedback. I changed the order in the distribution file. Sorry for missing this point.

**Pull request merged**
